ctx.go

```diff
@@ -817,6 +817,21 @@ func (c *DefaultCtx) Locals(key any, value ...any) any {
 	return value[0]
 }
 
+// Locals function utilizing Go's generics feature.
+// This function allows for manipulating and retrieving local values within a request context with a more specific data type.
+func Locals[V any](c Ctx, key any, value ...any) V {
+	ctx, ok := c.(*DefaultCtx)
+	if !ok {
+		panic(fmt.Errorf("failed to type-assert to *DefaultCtx"))
+	}
+	var v V
+	v, ok = ctx.Locals(key, value...).(V)
+	if !ok {
+		panic(fmt.Errorf("failed to type-assert to %T", v))
```

Comment:
would it be better to return an error instead of panicking?

Reply:
Maybe I think it could be changed with the default of its data type, like this:
```go
v, ok = c.Locals(key, value...).(V)
  if !ok {
    return v
  }
return v
```
What do you think?

Reply:
What do you think, if we return default value of variable and Locals Generic change to only get value from local not for set the value?

`c.Locals("name", "john")`
`fiber.Locals(c, "name")` // john
`fiber.Locals[string](c, "unknown") // <empty string>
`fiber.Locals[string](c, "unknown", "default") // default

so, we don't need for return an error, maybe like Query Generic?

Reply:
@nickajacks1 @efectn

Reply:
maybe like this:
![image](https://github.com/gofiber/fiber/assets/17839029/3119798c-00e9-4bc8-ab60-208d315198a3)

Reply:
```go
user := fiber.Locals(c, "user")
if user == User{} { /* stuff */ }

// or

user := fiber.Locals(c, "user", DefaultUser)
if user == DefaultUser { /* stuff */ }
```
vs
```go
user, err := fiber.Locals(c, "user")
if err != nil { /* stuff */ }
```
I think you're right that returning the zero value makes more sense given that there's a default value parameter.  Returning an error is a bit redundant with the default return value.

Reply:
Yeah returning zero-value seems better to me

what do you thing @gofiber/maintainers

Reply:
yes zero value please and without defaultValue
the method should also be usable as a setter and getter like the other method, so in any case I would expect that and no other pattern
so value should be the value you set and not defaultValue, otherwise we would have to rename the method